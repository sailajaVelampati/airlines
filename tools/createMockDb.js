/* eslint-disable no-console */
const fs = require("fs");
const path = require("path");
const mockData = require("./mockData");

const {
  newPassenger,
  flightsData,
  AncillaryServices,
  SeatNo,
  Meals,
  Shopping,
  flightAndPassengerDetails,
  AvalableFlights,
} = mockData;
const data = JSON.stringify({
  newPassenger,
  flightsData,
  AncillaryServices,
  SeatNo,
  Meals,
  Shopping,
  flightAndPassengerDetails,
  AvalableFlights,
});
const filepath = path.join(__dirname, "db.json");

fs.writeFile(filepath, data, function (err) {
  err ? console.log(err) : console.log("Mock DB created.");
});
