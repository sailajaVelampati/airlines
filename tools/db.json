{"newPassenger":{"seatNo":null,"name":"","ancillaryService":"","passport":false,"address":"","dob":false,"checked_in":false},"flightsData":[{"flightNo":"TY897","id":"TY897","time":1579917600000,"meals":["Vegetarian","Continetal","Indian","Non-Vegetarian"],"ancillaryServices":["wheel chair","infant"],"shopping":["Travel Kit","Eye Mask"],"from":"Delhi","to":"Hyderabad","seats":["A1","B1","A2","B2","C1","C2"]},{"flightNo":"NS023","id":"NS023","time":1579824000000,"meals":["Vegetarian","Non-Vegetarian"],"ancillaryServices":["wheel chair","infant"],"shopping":["Travel Kit","Power Bank"],"from":"Goa","to":"Bangalore","seats":["A1","B1","A2","B2","C1","C2"]},{"flightNo":"IP649","id":"IP649","time":1587686400000,"meals":["Vegetarian","Non-Vegetarian","Bengali"],"ancillaryServices":["wheel chair","infant"],"shopping":["Dairy","Power Bank"],"from":"Kolkata","to":"Chandighar","seats":["A1","B1","A2","B2","C1","C2"]}],"AncillaryServices":["wheel chair","infant"],"SeatNo":["A1","B1","A2","B2","C1","C2"],"Meals":["Vegetarian","Continetal","Indian","Non-Vegetarian","Indian","Italian","English"],"Shopping":["Travel Kit","Power Bank","Eye Mask","Ear Phones","Water Bottle","Dairy"],"flightAndPassengerDetails":{"flightNo":"TY897","id":"TY897","time":1579917600000,"meals":["Vegetarian","Continetal","Indian","Non-Vegetarian"],"ancillaryServices":["wheel chair","infant"],"shopping":["Travel Kit","Eye Mask"],"from":"Delhi","to":"Hyderabad","seats":["A1","B1","A2","B2","C1","C2"],"passengers":[{"name":"sailaja velampati","seatNo":"A1","passport":"k23892kl","DoB":"1989-12-04","address":"","ancillaryServices":["Infant"],"meals":[],"shopping":[],"checkIn":true},{"name":"Ram Kumar","seatNo":"A2","passport":"k2342kl","DoB":"1988-04-27","address":"","ancillaryServices":[],"meals":[],"shopping":[],"checkIn":true},{"name":"Achyuth Raghava","seatNo":"B2","passport":"","DoB":"2019-07-03","address":"","ancillaryServices":["Infant"],"meals":["Continental"],"shopping":[],"checkIn":false}]},"AvalableFlights":["TY897","NS023","IP649"]}