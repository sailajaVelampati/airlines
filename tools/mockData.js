const AvalableFlights = ["TY897", "NS023", "IP649"];
const SeatNo = ["A1", "B1", "A2", "B2", "C1", "C2"];
const AncillaryServices = ["wheel chair", "infant"];
const Meals = [
  "Vegetarian",
  "Continetal",
  "Indian",
  "Non-Vegetarian",
  "Indian",
  "Italian",
  "English",
];
const Shopping = [
  "Travel Kit",
  "Power Bank",
  "Eye Mask",
  "Ear Phones",
  "Water Bottle",
  "Dairy",
];
const flightsData = [
  {
    flightNo: AvalableFlights[0],
    id: AvalableFlights[0],
    time: 1579917600000,
    meals: ["Vegetarian", "Continetal", "Indian", "Non-Vegetarian"],
    ancillaryServices: AncillaryServices,
    shopping: ["Travel Kit", "Eye Mask"],
    from: "Delhi",
    to: "Hyderabad",
    seats: SeatNo,
  },
  {
    flightNo: AvalableFlights[1],
    id: AvalableFlights[1],
    time: 1579824000000,
    meals: ["Vegetarian", "Non-Vegetarian"],
    ancillaryServices: AncillaryServices,
    shopping: ["Travel Kit", "Power Bank"],
    from: "Goa",
    to: "Bangalore",
    seats: SeatNo,
  },
  {
    flightNo: AvalableFlights[2],
    id: AvalableFlights[2],
    time: 1587686400000,
    meals: ["Vegetarian", "Non-Vegetarian", "Bengali"],
    ancillaryServices: AncillaryServices,
    shopping: ["Dairy", "Power Bank"],
    from: "Kolkata",
    to: "Chandighar",
    seats: SeatNo,
  },
];

const passengers = [
  {
    name: "sailaja velampati",
    seatNo: "A1",
    passport: "k23892kl",
    DoB: "1989-12-04",
    address: "",
    ancillaryServices: ["Infant"],
    meals: [],
    shopping: [],
    checkIn: true,
  },
  {
    name: "Ram Kumar",
    seatNo: "A2",
    passport: "k2342kl",
    DoB: "1988-04-27",
    address: "",
    ancillaryServices: [],
    meals: [],
    shopping: [],
    checkIn: true,
  },
  {
    name: "Achyuth Raghava",
    seatNo: "B2",
    passport: "",
    DoB: "2019-07-03",
    address: "",
    ancillaryServices: ["Infant"],
    meals: ["Continental"],
    shopping: [],
    checkIn: false,
  },
];
const flightAndPassengerDetails = {
  ...flightsData[0],
  passengers,
};

const newPassenger = {
  seatNo: null,
  name: "",
  ancillaryService: "",
  passport: false,
  address: "",
  dob: false,
  checked_in: false,
};

// Using CommonJS style export so we can consume via Node (without using Babel-node)
module.exports = {
  newPassenger,
  flightsData,
  AncillaryServices,
  SeatNo,
  Meals,
  Shopping,
  flightAndPassengerDetails,
  AvalableFlights,
};
