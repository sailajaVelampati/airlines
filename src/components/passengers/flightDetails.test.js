import React from "react";
import FlightDetails from "./flightDetails";
import renderer from "react-test-renderer";
import { flightAndPassengerDetails } from "../../../tools/mockData";

describe.only("Flight and Passenger Details", () => {
  it("should Load only flight select dropdown when data is empty", () => {
    const tree = renderer.create(
      <FlightDetails data={{}} handleFlightSelectChange={jest.fn()} />
    );
    expect(tree).toMatchSnapshot();
  });

  it("should Load flight details along with flight select dropdown when data is empty", () => {
    const tree = renderer.create(
      <FlightDetails
        data={flightAndPassengerDetails}
        handleFlightSelectChange={jest.fn()}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
