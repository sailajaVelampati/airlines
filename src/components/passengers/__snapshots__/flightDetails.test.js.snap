// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight and Passenger Details should Load flight details along with flight select dropdown when data is empty 1`] = `
Array [
  <nav
    aria-label="contacts"
    className="MuiList-root makeStyles-FlightDetailsListStyle-5 MuiList-padding"
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 16v-2l-8-5V3.5c0-.83-.67-1.5-1.5-1.5S10 2.67 10 3.5V9l-8 5v2l8-2.5V19l-2 1.5V22l3.5-1 3.5 1v-1.5L13 19v-5.5l8 2.5z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          TY897
        </span>
      </div>
    </div>
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
          <path
            d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          1/25/2020, 7:30:00 AM
        </span>
      </div>
    </div>
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M2.5 19h19v2h-19v-2zm19.57-9.36c-.21-.8-1.04-1.28-1.84-1.06L14.92 10l-6.9-6.43-1.93.51 4.14 7.17-4.97 1.33-1.97-1.54-1.45.39 2.59 4.49s7.12-1.9 16.57-4.43c.81-.23 1.28-1.05 1.07-1.85z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Delhi
        </span>
      </div>
    </div>
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M2.5 19h19v2h-19v-2zm16.84-3.15c.8.21 1.62-.26 1.84-1.06.21-.8-.26-1.62-1.06-1.84l-5.31-1.42-2.76-9.02L10.12 2v8.28L5.15 8.95l-.93-2.32-1.45-.39v5.17l16.57 4.44z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Hyderabad
        </span>
      </div>
    </div>
  </nav>,
  <div
    className="MuiFormControl-root makeStyles-formControl-1"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      id="demo-mutiple-chip-label"
    >
      Select Flight
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="demo-mutiple-chip-label demo-mutiple-chip"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="demo-mutiple-chip"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        id="select-multiple-chip"
        onAnimationStart={[Function]}
        type="hidden"
        value=""
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>,
]
`;

exports[`Flight and Passenger Details should Load only flight select dropdown when data is empty 1`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink={false}
    id="demo-mutiple-chip-label"
  >
    Select Flight
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="demo-mutiple-chip-label demo-mutiple-chip"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="demo-mutiple-chip"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      id="select-multiple-chip"
      onAnimationStart={[Function]}
      type="hidden"
      value=""
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
